File: infraestructrure/utils/OpenSocialMedia.ts
import { Alert, Linking } from "react-native";

export const openSocialMedia = async (url: string) => {
  const supported = await Linking.canOpenURL(url);

  if (supported) {
    return await Linking.openURL(url);
  } else {
    Alert.alert("Error", "No se pudo abrir el enlace");
  }
};

File: infraestructrure/utils/OpenSocialMedia.swift
import SwiftUI

func openSocialMedia(url: String) {
    guard let link = URL(string: url), UIApplication.shared.canOpenURL(link) else {
        showAlert(title: "Error", message: "No se pudo abrir el enlace")
        return
    }

    UIApplication.shared.open(link, options: [:], completionHandler: nil)
}

func showAlert(title: String, message: String) {
    let alert = UIAlertController(title: title, message: message, preferredStyle: .alert)
    alert.addAction(UIAlertAction(title: "OK", style: .default))
    if let topController = UIApplication.shared.windows.first?.rootViewController {
        topController.present(alert, animated: true)
    }
}
